frontend-react/src/resources/SubmissionsResource.ts

```diff
@@ -8,7 +8,9 @@ export default class SubmissionsResource extends AuthResource {
     readonly warningCount: number = 0;
 
     pk(params: any) {
-        return this.id;
+        // For failed submissions, the report id will be null. Rest Hooks will not persist a record without a pk, thus
+        // falling back to using createdAt.
+        return this.id || this.createdAt?.toString();
```

Comment:
@kevinhaube @TomNUSDS This was the solution to the null submission problem. Report ids are not guaranteed to be non-null from the server. Rest Hooks will no persist records with null pks so I'm falling back to using the createdAt string, which as far as I can tell will always be non-null.

Reply:
I never got a notification for this.. I wonder why?

Another solution is to supply a default in the "Resource" loader. [See ](https://github.com/CDCgov/prime-reportstream/blob/97b88eaa8cd0d03ead08b336a04a4271822d0f1d/frontend-react/src/resources/OrgSenderSettingsResource.ts#L24)

Reply:
If multiple items use the default pk, i wonder if there would be caching problems? createAt is kinda a hack, but is likely never to collide